### ISSUE TITLE
Should base image tag be linted for `latest`

### ISSUE THREAD

**Issue by TheYorkshireDev** [replicatedhq/dockerfilelint#98]
Hi,

I was experimenting with linting some dockerfiles and noticed for multi-stage builds the base image we use generally as a builder, does not get linted for the `:latest` tag. Is this intentional?

For Example:
```
FROM golang:latest as builder
WORKDIR /go/src/github.com/alexellis/href-counter/
RUN go get -d -v golang.org/x/net/html
COPY app.go .
RUN CGO_ENABLED=0 GOOS=linux go build -a -installsuffix cgo -o app .

FROM alpine:latest
RUN apk add ca-certificates
WORKDIR /root/
copy --from=builder /go/src/github.com/alexellis/href-counter/app .
CMD ["./app"]
```

The second `FROM` gets the warning, but the first does not.

Thanks

**Comment by prasenjithaty** [comment]
I guess the issue is that the linter does not support Docker multi-stage builds yet introduced in Docker 17.05 and errors out with:
```
The first instruction in a Dockerfile must specify the base image using a FROM command. Additionally, FROM cannot appear later in a Dockerfile.

**Comment by alexellis** [comment]
How about now?

**Comment by mildebrandt** [comment]
I think this was fixed too well. It does warn on both `latest` tags in the `FROM` statements....but it doesn't like the reference to a previous stage.

I changed the Dockerfile to this:
```
FROM golang:latest as builder
WORKDIR /go/src/github.com/alexellis/href-counter/
RUN go get -d -v golang.org/x/net/html
COPY app.go .
RUN CGO_ENABLED=0 GOOS=linux go build -a -installsuffix cgo -o app .

FROM builder
RUN ls -l /

FROM alpine:latest
RUN apk add ca-certificates
WORKDIR /root/
copy --from=builder /go/src/github.com/alexellis/href-counter/app .
CMD ["./app"]
```

And I get the following messages:
```
$ docker run --rm -it -v $PWD/Dockerfile:/Dockerfile replicated/dockerfilelint /Dockerfile

File:   /Dockerfile
Issues: 5

Line 1: FROM golang:latest as builder
Issue  Category      Title                 Description
    1  Clarity       Base Image Latest     Base images should not use the latest tag.
                     Tag

Line 7: FROM builder
Issue  Category      Title                 Description
    2  Clarity       Base Image Missing    Base images should specify a tag to use.
                     Tag

Line 10: FROM alpine:latest
Issue  Category      Title                 Description
    3  Clarity       Base Image Latest     Base images should not use the latest tag.
                     Tag

Line 11: RUN apk add ca-certificates
Issue  Category      Title                 Description
    4  Optimization  Consider `--no-cache  Consider using a `--no-cache` (supported in alpine linux >= 3.3) or
                     or --update with rm   `--update` followed by the command `rm -rf /var/cache/apk/*` when
                     -rf                   `apk` adding packages.  This will result in a smaller image size.
                     /var/cache/apk/*`     For
                                           more information, see [this
                                           link](https://github.com/gliderlabs/docker-alpine/blob/master/docs/
                                           usage.md)

Line 13: copy --from=builder /go/src/github.com/alexellis/href-counter/app .
Issue  Category      Title                 Description
    5  Clarity       Capitalize            For clarity and readability, all instructions in a Dockerfile
                     Dockerfile            should be uppercase.
                     Instructions          This is a convention adopted by most of the official images and
                                           greatly improves readability in long Dockerfiles.  For an example
                                           of
                                           why this makes a difference, check out the current [redis
                                           Dockerfile](https://github.com/docker-library/redis/blob/b375650fb6
                                           9b7db819e90c0033433c705b28656e/3.0/Dockerfile)
                                           and you should be able to easily see the instructions used.
```

I would not expect the `Line 7: FROM builder` message since it's referencing a previous stage.
